nexus/src/app/sagas/region_replacement_drive.rs

```diff
@@ -1284,11 +1284,12 @@ async fn execute_pantry_drive_action(
     volume_id: Uuid,
     job_id: Uuid,
 ) -> Result<(), ActionError> {
-    // Importantly, _do not use `call_pantry_attach_for_disk`_! That uses
-    // `retry_until_known_result`, which we _do not want here_. The Pantry
-    // attach can fail if there's a racing Volume checkout to be sent to
-    // Propolis. Additionally, that call uses `attach` instead of
-    // `attach_activate_background`, which means it will hang on the activation.
+    // Importantly, _do not use `call_pantry_attach_for_disk`_! That retries
+    // as long as `pantry_address` is still resolvable in DNS, which we _do not
+    // want here_. The Pantry attach can fail if there's a racing Volume
+    // checkout to be sent to Propolis. Additionally, that call uses `attach`
+    // instead of `attach_activate_background`, which means it will hang on the
+    // activation.
```

Comment:
I would update this message to only

```
Importantly, _do not use `call_pantry_attach_for_disk`_! That call uses `attach` instead of `attach_activate_background`, which means it will hang on the activation.
```

At some point, some sort of retry should be added here, but that shouldn't block this PR.

Reply:
Updated in [fb21fb9](https://github.com/oxidecomputer/omicron/pull/6866/commits/fb21fb945740a624a0d93ddeda5970f683bd43a1)